# Multiple Linear Regression on real estate data(with and without dummy variables)
This notebook walks through two versions of a multiple linear regression on the same real estate problem:

1. **Part 1:** Regression using `size` and `year` only.
2. **Part 2:** Regression using `size`, `year`, and `view` (with `view` converted into a dummy variable).

Comparing the two shows how adding a well-chosen categorical variable (encoded as a dummy) can improve model fit.

## Part 1: Regression with `size` and `year`

## Import the relevant libraries

In [3]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

## Load the data 

In [79]:
data = pd.read_csv('real_estate_price_size_year.csv')
pd.set_option('display.max_rows', 10)

In [80]:
data 

,price,size,year
0,234314.144,643.09,2015
1,228581.528,656.22,2009
2,281626.336,487.29,2018
3,401255.608,1504.75,2015
4,458674.256,1275.46,2009
...,...,...,...
95,252460.400,549.80,2009
96,310522.592,1037.44,2009
97,383635.568,1504.75,2006
98,225145.248,648.29,2015


In [81]:
data.describe()

,price,size,year
count,100.000000,100.000000,100.000000
mean,292289.470160,853.024200,2012.600000
std,77051.727525,297.941951,4.729021
min,154282.128000,479.750000,2006.000000
25%,234280.148000,643.330000,2009.000000
50%,280590.716000,696.405000,2015.000000
75%,335723.696000,1029.322500,2018.000000
max,500681.128000,1842.510000,2018.000000


## Create the regression

### Declare the dependant and independant variables

In [7]:
y = data ['price']
x1 = data [['size', 'year']]

### Regression

In [8]:
x = sm.add_constant(x1)
results = sm.OLS(y,x).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  price   R-squared:                       0.776
Model:                            OLS   Adj. R-squared:                  0.772
Method:                 Least Squares   F-statistic:                     168.5
Date:                Mon, 31 Aug 2026   Prob (F-statistic):           2.77e-32
Time:                        18:24:25   Log-Likelihood:                -1191.7
No. Observations:                 100   AIC:                             2389.
Df Residuals:                      97   BIC:                             2397.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -5.772e+06   1.58e+06     -3.647      0.000   -8.91e+06   -2.63e+06
size         227.7009     12.474     18.254      0.000     202.943     252.458
year        2916.7853    785.896      3.711      0.000    1357.000    4476.571
==============================================================================
Omnibus:                       10.083   Durbin-Watson:                   2.250
Prob(Omnibus):                  0.006   Jarque-Bera (JB):                3.678
Skew:                           0.095   Prob(JB):                        0.159
Kurtosis:                       2.080   Cond. No.                     9.41e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 9.41e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

## Part 2: Regression with `size`, `year`, and `view` (dummy variable)

## Load the data 

In [73]:
more_data = pd.read_csv('real_estate_price_size_year_view.csv')
pd.set_option('display.max_rows', 10)

In [74]:
more_data

,price,size,year,view
0,234314.144,643.09,2015,No sea view
1,228581.528,656.22,2009,No sea view
2,281626.336,487.29,2018,Sea view
3,401255.608,1504.75,2015,No sea view
4,458674.256,1275.46,2009,Sea view
...,...,...,...,...
95,252460.400,549.80,2009,Sea view
96,310522.592,1037.44,2009,No sea view
97,383635.568,1504.75,2006,No sea view
98,225145.248,648.29,2015,No sea view


In [75]:
data = more_data.copy()

In [76]:
data['view'] = data['view'].map({'No sea view' : 1 , 'Sea view' : 0})

In [77]:
data

,price,size,year,view
0,234314.144,643.09,2015,1
1,228581.528,656.22,2009,1
2,281626.336,487.29,2018,0
3,401255.608,1504.75,2015,1
4,458674.256,1275.46,2009,0
...,...,...,...,...
95,252460.400,549.80,2009,0
96,310522.592,1037.44,2009,1
97,383635.568,1504.75,2006,1
98,225145.248,648.29,2015,1


In [78]:
data.describe(include='all')

,price,size,year,view
count,100.000000,100.000000,100.000000,100.000000
mean,292289.470160,853.024200,2012.600000,0.510000
std,77051.727525,297.941951,4.729021,0.502418
min,154282.128000,479.750000,2006.000000,0.000000
25%,234280.148000,643.330000,2009.000000,0.000000
50%,280590.716000,696.405000,2015.000000,1.000000
75%,335723.696000,1029.322500,2018.000000,1.000000
max,500681.128000,1842.510000,2018.000000,1.000000


## Create the regression
### Declare the dependant and independant variables

In [71]:
p = data['price']
q1 = data[['size','year','view']]

## regression

In [72]:
q = sm.add_constant(q1)
results = sm.OLS(p,q).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  price   R-squared:                       0.913
Model:                            OLS   Adj. R-squared:                  0.910
Method:                 Least Squares   F-statistic:                     335.2
Date:                Mon, 31 Aug 2026   Prob (F-statistic):           1.02e-50
Time:                        21:07:21   Log-Likelihood:                -1144.6
No. Observations:                 100   AIC:                             2297.
Df Residuals:                      96   BIC:                             2308.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -5.341e+06   9.94e+05     -5.373      0.000   -7.31e+06   -3.37e+06
size         223.0316      7.838     28.455      0.000     207.473     238.590
year        2718.9489    493.502      5.510      0.000    1739.356    3698.542
view       -5.673e+04   4627.695    -12.258      0.000   -6.59e+04   -4.75e+04
==============================================================================
Omnibus:                       29.224   Durbin-Watson:                   1.965
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               64.957
Skew:                           1.088   Prob(JB):                     7.85e-15
Kurtosis:                       6.295   Cond. No.                     9.42e+05
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 9.42e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

## Comparison & Takeaway

| Model | R-squared | Adj. R-squared | Notes |
|---|---|---|---|
| Part 1: size + year | 0.776 | 0.772 | Two numeric predictors only |
| Part 2: size + year + view | 0.913 | 0.910 | Adding the `view` dummy substantially improves fit |

Adding the `view` dummy variable increased R-squared from **0.776 to 0.913**, showing that whether a property has a sea view is a meaningful, statistically significant predictor of price (coefficient ≈ $56,730, p = 0.000).